### DISCUSSION TITLE
uWebSockets -vs- LibWebSocket (LWS) Benchmark

### DISCUSSION THREAD

**Discussion by pkrumins** [uNetworking/uWebSockets#1503] [General]
Hi allz,

Has anyone benchmarked uWebSocket vs LibWebSocket? The benchmarks for uWebSocket are impressive but how does it compare to LibWebSocket?

Thx!

**Comment by uNetworkingAB** [comment]
Uws began as a wrapper for libwebsockets back in 2015, but after committing a 10x performance improvement to libwebsocket's parser (it used to do 1 byte per state machine lookup) I figured I could write such a server better from scratch. So in 2016 uws was released

**Reply by pkrumins** [reply]
Thanks for your response!

**Comment by uNetworkingAB** [comment]
Here is a link to a kept fork of the original https://github.com/NodeCedu/node-lws

I guess it started in January of 2016, not 2015
